test/data/testSP.py

```diff
@@ -255,20 +255,16 @@ def testmoments(self):
                 self.assertEquals(m, 0.0)
 
     @skipForParser('ADF', 'Does not support metadata yet')
-    @skipForParser('GAMESS', 'Does not support metadata yet')
     @skipForParser('GAMESSUK', 'Does not support metadata yet')
-    @skipForParser('Gaussian', 'Does not support metadata yet')
-    @skipForParser('Jaguar', 'Does not support metadata yet')
     @skipForParser('Molcas','The parser is still being developed so we skip this test')
-    @skipForParser('Molpro', 'Does not support metadata yet')
-    @skipForParser('NWChem', 'Does not support metadata yet')
-    @skipForParser('ORCA', 'Does not support metadata yet')
-    @skipForParser('Psi', 'Does not support metadata yet')
-    @skipForParser('QChem', 'Does not support metadata yet')
     def testmetadata(self):
         """Does metadata have expected keys and values?"""
         self.assertTrue(hasattr(self.data, "metadata"))
-        self.assertIn("basis_set", self.data.metadata)
+        if self.logfile.logname not in ['ORCA', 'Psi']:
+            self.assertIn("basis_set", self.data.metadata)
+        if self.logfile.logname == 'ORCA':
+            self.assertIn("input_file_name", self.data.metadata)
+            self.assertIn("input_file_contents", self.data.metadata)

```

Comment:
This isn't maintainable, so our next step can be to move metadata testing to its own file and have a test for each field.